src/main/java/org/springframework/integration/aws/outbound/KinesisMessageHandler.java

```diff
@@ -245,4 +297,44 @@ private PutRecordRequest buildPutRecordRequest(Message<?> message) {
 				.withData(data);
 	}
 
+	@SuppressWarnings("rawtypes")
+	private AsyncHandler<? extends AmazonWebServiceRequest, ?> getAsyncHandler(final Message<?> message,
+																			final AmazonWebServiceRequest request) {
+		if (this.asyncHandler != null) {
+			return this.asyncHandler;
```

Comment:
No, this always must be delegated from our internal one. I somehow think that would be better to let end-user to provide his onw impl and configure failure and success channels.

Reply:
I'm not understanding what you mean by always delegating from our internal one. What should happen when `this.asyncHandler` is not null? What should happen when it is null?

Reply:
Are you saying we shouldn't provide our own implementation of `AsyncHandler` or expose channels at all and force clients of this class to provide their own implementation using channels?